### PR TITLE
Revert "Part 3 of shim for running gcloud storage: Translate top level flags and headers"

### PR DESCRIPTION
**Pull request by dilipped**
Reverts GoogleCloudPlatform/gsutil#1419

I accidentally merged the submodule changes that were not related to this PR. I will send out a separate PR to fix it.

### CODE REVIEW

File: gslib/tests/test_shim_util.py [hunk 1]
```diff
@@ -360,301 +360,6 @@ def test_print_gcloud_storage_env_vars_in_dry_run_mode(self):
...
-        'Cache-Control': 'fake_cache_cantrol',
-        'Content-Disposition': 'fake_Content-Disposition',
-        'Content-Encoding': 'fake_Content-Encoding',
-        'Content-Language': 'fake_Content-Language',
-        'Content-Type': 'fake_Content-Type',
-        'Content-MD5': 'fake_Content-MD5',
-        'custom-time': 'fake_time',
-        # Precondition headers
-        'x-goog-if-generation-match': 'fake_gen_match',
-        'x-goog-if-metageneration-match': 'fake_gen_match',
```
Comment (NickGoog) [inline gslib/tests/test_shim_util.py:513]: 'fake_metagen_match"

File: gslib/tests/test_shim_util.py [hunk 2]
```diff
@@ -360,301 +360,6 @@ def test_print_gcloud_storage_env_vars_in_dry_run_mode(self):
...
-    self.assertCountEqual(flags, [
-        '--cache-control=fake_cache_cantrol',
-        '--content-disposition=fake_Content-Disposition',
-        '--content-encoding=fake_Content-Encoding',
-        '--content-language=fake_Content-Language',
-        '--content-type=fake_Content-Type',
-        '--content-md5=fake_Content-MD5',
-        '--custom-time=fake_time',
-        '--if-generation-match=fake_gen_match',
-        '--if-metageneration-match=fake_gen_match',
```
Comment (NickGoog) [inline gslib/tests/test_shim_util.py:529]: 'fake_metagen_match'

File: gslib/tests/test_shim_util.py [hunk 3]
```diff
@@ -360,301 +360,6 @@ def test_print_gcloud_storage_env_vars_in_dry_run_mode(self):
...
-      self._fake_command._translate_headers()
-
-  @mock.patch.object(shim_util,
-                     'PRECONDITONS_ONLY_SUPPORTED_COMMANDS',
-                     new={'fake_shim'})
-  def test_translate_headers_for_precondition_supported_command_with_valid_header(
-      self):
-    self._fake_command.headers = {
-        'x-goog-if-generation-match': 'fake_gen_match',
-        'x-goog-if-metageneration-match': 'fake_gen_match',
```
Comment (NickGoog) [inline gslib/tests/test_shim_util.py:553]: 'fake_metagen_match'

File: gslib/tests/test_shim_util.py [hunk 4]
```diff
@@ -360,301 +360,6 @@ def test_print_gcloud_storage_env_vars_in_dry_run_mode(self):
...
-        # Custom metadata. These should be ignored.
-        'x-goog-meta-foo': 'fake_goog_meta',
-    }
-    flags = self._fake_command._translate_headers()
-    self.assertCountEqual(flags, [
-        '--if-generation-match=fake_gen_match',
-        '--if-metageneration-match=fake_gen_match',
```
Comment (NickGoog) [inline gslib/tests/test_shim_util.py:560]: 'fake_metagen_match'

File: gslib/tests/test_shim_util.py [hunk 5]
```diff
@@ -360,301 +360,6 @@ def test_print_gcloud_storage_env_vars_in_dry_run_mode(self):
...
-        'x-goog-api-version': '2',  # Should be ignored.
-    }
-    self.assertEqual(self._fake_command._translate_headers(),
-                     ['--if-generation-match=fake_gen_match'])
-
-
-class TestGetFlagFromHeader(testcase.GsUtilUnitTestCase):
-  """Test Command.get_flag_from_header function.
-  
-  We only test the unset functionality, because rest of the workflows have been
```
Comment (NickGoog) [inline gslib/tests/test_shim_util.py:603]: No comma before "because."
More details: https://www.grammarly.com/blog/comma-before-because/

**Review by NickGoog — APPROVED** [review]
Make sure the CI failures are unrelated, please

**Comment by dilipped** [comment]
Will address the comment in the revert of this revert :P 

The test failures are because of a separate github CI issue being fixed here: https://github.com/GoogleCloudPlatform/gsutil/pull/1443

**Pull request merged**